src/apme_engine/daemon/health_check.py

```
@@ -61,64 +89,116 @@ def check_grpc_health(
         req = common_pb2.HealthRequest()
         resp = stub.Health(req, timeout=timeout)
-        channel.close()
+    except grpc.RpcError as e:
+        elapsed_ms = (time.perf_counter() - start) * 1000
+        return {
+            "ok": False,
+            "status": None,
+            "error": e.details() or str(e.code()),
+            "latency_ms": round(elapsed_ms, 2),
+        }
+    except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)